doc(deps): Add Version Support Policy (#3466)
argoproj/argo-helm: README.md

helm repo add argo https://argoproj.github.io/argo-helm
```

## Version Support Policy
As our project is maintained by a small team, we must focus our limited resources on following upstream projects and ensuring the stability of the latest version.

Consequently, **we do not provide bug fixes or security patches for older versions.** Our official support is limited to **the latest version of the upstream projects** only.

> We strongly encourage all users to upgrade to the latest version to benefit from the most recent features, bug fixes, and security patches.

Reviewer: Maybe just a sentence at the end calling for others to join the team, link to the argoproj membership process? 

Other than that, I think this is good.

Author: I added a commit for them. 🙏 
https://github.com/argoproj/argo-helm/pull/3466/commits/cbb122aefc2fbe49d3d072424d798358b837d984

helm repo add argo https://argoproj.github.io/argo-helm
```

## Version Support Policy
As our project is maintained by a small team, we must focus our limited resources on following upstream projects and ensuring the stability of the latest version.

Consequently, **we do not provide bug fixes or security patches for older versions.** Our official support is limited to **the latest version of the upstream projects** only.

We strongly encourage all users to upgrade to the latest version to benefit from the most recent features, bug fixes, and security patches.

### For Users Unable to Upgrade
If you are unable to upgrade to the latest version due to specific constraints, please follow the below to patch.

> 1. Please upgrade Helm Chart to the latest version for your minor version. e.g: If you used `v8.2.0`, please update to `v8.2.6`, the latest version of `v8.2.x`.

Author: I fixed them in https://github.com/argoproj/argo-helm/pull/3466/commits/e5b75a051de3ee5e9499b3b193d00790f2055519 . 🙏

2. Please override the image tag (`.global.image.tag`) to use a specific version.

### How You Can Help

> This policy may evolve as our team grows. If you are interested in joining our team and helping us expand our support capabilities, we encourage you to read the [Community Membership](https://github.com/argoproj/argoproj/blob/main/community/membership.md).

Author: I thought I was supposed to change the URL..?
https://github.com/argoproj/argo-helm/pull/3466#discussion_r2325334573

helm repo add argo https://argoproj.github.io/argo-helm
```

## Version Support Policy
As our project is maintained by a small team, we must focus our limited resources on following upstream projects and ensuring the stability of the latest version.

Consequently, **we do not provide bug fixes or security patches for older versions.** Our official support is limited to **the latest version of the upstream projects** only.

We strongly encourage all users to upgrade to the latest version to benefit from the most recent features, bug fixes, and security patches.

### For Users Unable to Upgrade
> **Warning:**
> This doesn't work all the time. We strongly recommend upgrading Helm Chart to the latest version.

If you are unable to upgrade to the latest version due to specific constraints, please follow the below to patch.

1. Upgrade Helm Chart to the latest version for your minor version. e.g: If you used `v8.2.0`, update to `v8.2.6`, the latest version of `v8.2.x`.
2. Override the image tag (`.global.image.tag`) to use a specific version.

### How You Can Help

> This policy may evolve as our team grows. If you are interested in joining our team and helping us expand our support capabilities, we encourage you to read the [Community Membership](https://github.com/argoproj/argoproj/blob/main/community/membership.md) for details.

Reviewer: ```suggestion
...we encourage you to read the [Community Membership Guide](https://github.com/argoproj/argoproj/blob/main/community/membership.md) for details.
```

helm repo add argo https://argoproj.github.io/argo-helm
```

## Version Support Policy
As our project is maintained by a small team, we must focus our limited resources on following upstream projects and ensuring the stability of the latest version.

Consequently, **we do not provide bug fixes or security patches for older versions.** Our official support is limited to **the latest version of the upstream projects** only.

We strongly encourage all users to upgrade to the latest version to benefit from the most recent features, bug fixes, and security patches.

### For Users Unable to Upgrade
If you are unable to upgrade to the latest version due to specific constraints, you can manually override the image tag to use a specific version.

Example for [argo-cd](https://github.com/argoproj/argo-helm/blob/15658c3c0aeaa2f8fc5ef77555c78e5ce2256e12/charts/argo-cd/values.yaml#L62):
```yaml
global:
  image:
    tag: v3.1.3
```

### How You Can Help

> This policy may evolve as our team grows. If you are interested in becoming a maintainer and helping us expand our support capabilities, we encourage you to read our [contribution guidelines](CONTRIBUTING.md).

Reviewer: given that link is a line below, and you mention becoming a maintainer, I'd link here: https://github.com/argoproj/argoproj/blob/main/community/membership.md

... which helpfully doesn't include our project.

Author: Wow I didn't know it... Thank you! will fix it.
